### DISCUSSION TITLE
Enable zstd compression by default for built packages?

### DISCUSSION THREAD

**Discussion by dralley** [rpm-rs/rpm#127] [Polls]
Currently the default is to use no compression for packages built using this library, but that possibly isn't a good default.  Zstd compression is the default for `rpmbuild`, and compressed RPMs can save a significant amount of disk space and network bandwidth.

There were some numbers produced a few years ago by Fedora: https://fedoraproject.org/wiki/Changes/Switch_RPMs_to_zstd_compression

As you can see even lower levels of zstd compression (which are extremely cheap) would would dramatically reduce package size over uncompressed packages.  However it is suggested that for Fedora zstd level 19 was still the correct choice since the build time would be significantly dominated by infrastructure and distribution costs.

Pros:

* Smaller packages
* Default would match `rpmbuild`

Cons:

* If we use a high compression level (as Zstd is currently configured to do in rpm-rs), then builds may take a fair bit longer
* Requires a zstd dependency (not a big deal IMO as for the purpose of reading RPMs you likely need this regardless)

**Comment by dralley** [comment]
@cmeister2 @drahnr

**Comment by drahnr** [comment]
I would make it opt in, but use compression in all examples given.
